src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.Comments.cs

```diff
@@ -9,40 +9,53 @@ namespace System.IO.Compression.Tests
     public partial class zip_CreateTests : ZipFileTestBase
     {
         [Theory]
-        // General purpose bit flag must get the appropriate bit set if a file comment or an entry name is unicode
-        [InlineData("ascii", "ascii!!!", "utf-8", "UÄäÖöÕõÜü")]
-        [InlineData("utf-8", "UÄäÖöÕõÜü", "ascii", "ascii!!!")]
-        [InlineData("ascii", "ascii!!!", "latin1", "LÄäÖöÕõÜü")]
-        [InlineData("latin1", "LÄäÖöÕõÜü", "ascii", "ascii!!!")]
-        [InlineData("utf-8", "UÄäÖöÕõÜü", "latin1", "LÄäÖöÕõÜü")]
-        [InlineData("latin1", "LÄäÖöÕõÜü", "utf-8", "UÄäÖöÕõÜü")]
-        public static void Create_ZipArchiveEntry_DifferentEncodings_FullName_And_Comment(string encodingName1, string text1, string encodingName2, string text2)
-        {
-            Encoding encoding1 = Encoding.GetEncoding(encodingName1);
-            Encoding encoding2 = Encoding.GetEncoding(encodingName2);
-            string entryName = encoding1.GetString(encoding1.GetBytes(text1));
-            string comment = encoding2.GetString(encoding2.GetBytes(text2));
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_NullEncoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, null);
+
+        [Theory]
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_Utf8Encoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, Encoding.UTF8);
 
-            var stream = new MemoryStream();
-            var testStream = new WrappedStream(stream, true, true, true, null);
+        [Theory]
+        [MemberData(nameof(Latin1Comment_Data))]
+        public static void Create_Comment_AsciiEntryName_Latin1Encoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal("file.txt", originalComment, expectedComment, Encoding.Latin1);
 
-            // Create with no encoding to autoselect it if one of the two strings is unicode
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true))
+        [Theory]
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Create_Comment_Utf8EntryName_NullEncoding(string originalComment, string expectedComment) =>
+            Create_Comment_EntryName_Encoding_Internal($"{SmileyEmoji}.txt", originalComment, expectedComment, null);
```

Comment:
Does the entry name really need to be tested? It is not something that is being added in this PR.

Reply:
The entry name is specifically being affected by this PR, because both the comment and the entry name share the same encoding the user specifies.

Reply:
I was expecting that there were already tests that covered `EntryName` with multiple encodings, if that's not the case then it is fine.